storage/src/test/java/tech/pegasys/teku/storage/client/CombinedChainDataClientTest.java

```diff
@@ -146,6 +150,79 @@ void getsEarliestAvailableBlobSidecarSlot() {
     assertThat(result).hasValue(UInt64.ONE);
   }
 
+  @Test
+  void getStateForBlockProduction_directsToStateAtSlotExact()
+      throws ExecutionException, InterruptedException {
+    final BeaconState state = dataStructureUtil.randomBeaconState(UInt64.valueOf(2));
+    final Optional<Bytes32> recentBlockRoot =
+        Optional.of(spec.getBlockRootAtSlot(state, UInt64.ONE));
+    when(recentChainData.getBlockRootInEffectBySlot(UInt64.ONE)).thenReturn(recentBlockRoot);
+    when(recentChainData.getStore()).thenReturn(store);
+    when(store.retrieveStateAtSlot(any()))
+        .thenReturn(SafeFuture.completedFuture(Optional.of(state)));
+    final SafeFuture<Optional<BeaconState>> future =
+        client.getStateForBlockProduction(UInt64.ONE, false);
+    assertThat(future.get()).contains(state);
+  }
+
+  @Test
+  void getStateForBlockProduction_whenEnabeldAndHaveNoBestBlockRoot()
+      throws ExecutionException, InterruptedException {
+    final BeaconState state = dataStructureUtil.randomBeaconState(UInt64.valueOf(2));
+    final Optional<Bytes32> recentBlockRoot =
+        Optional.of(spec.getBlockRootAtSlot(state, UInt64.ONE));
+    when(recentChainData.getStore()).thenReturn(store);
+
+    when(recentChainData.getBestBlockRoot()).thenReturn(Optional.empty());
+    when(recentChainData.getBlockRootInEffectBySlot(UInt64.ONE)).thenReturn(recentBlockRoot);
+    when(store.retrieveStateAtSlot(any()))
+        .thenReturn(SafeFuture.completedFuture(Optional.of(state)));
+
+    final SafeFuture<Optional<BeaconState>> future =
+        client.getStateForBlockProduction(UInt64.ONE, true);
+    assertThat(future.get()).contains(state);
+  }
+
+  @Test
+  void getStateForBlockProduction_whenEnabeldAndBestBlockRootMatches()
+      throws ExecutionException, InterruptedException {
+    final BeaconState state = dataStructureUtil.randomBeaconState(UInt64.valueOf(2));
+    final Optional<Bytes32> recentBlockRoot =
+        Optional.of(spec.getBlockRootAtSlot(state, UInt64.ONE));
+    when(recentChainData.getStore()).thenReturn(store);
+
+    when(recentChainData.getBestBlockRoot()).thenReturn(recentBlockRoot);
+    when(recentChainData.getProposerHead(any(), any())).thenReturn(recentBlockRoot.get());
+    when(recentChainData.getBlockRootInEffectBySlot(UInt64.ONE)).thenReturn(recentBlockRoot);
```

Comment:
It's not verified here and in `getStateForBlockProduction_whenEnabeldAndHaveNoBestBlockRoot` that we actually walked this path. Adding `verify` could be beneficial

Reply:
added verifies to test cases.